### DISCUSSION TITLE
polypropylene parameterization

### DISCUSSION THREAD

**Discussion by shakhawath** [marrink-lab/polyply_1.0#359] [Martini3]
Hello, I was wondering if there is any parameterization done for polypropylene using Martini3. Also, will it be possible to use polyply to obtain a combined PEO:PPO polymer structure and itp file?

Thanks.

**Comment by fgrunewald** [comment]
Hi @shakhawath, 

The Martini3 PPO (I assume you mean polypropylene oxide) model is described [here](https://pure.rug.nl/ws/portalfiles/portal/756637593/Complete_thesis.pdf) in Chapter 5. Unfortunately, we are still in the process of consolidating all the files and uploading them to the library. Below you find the most recent PPO version. Simply, past it into a file called PPO.ff and generate the homopolymer using:
```
polyply gen_params -f PPO.ff -seq PPO:10 -o test.itp -name PPO
```
To make the co-polymer get the PEO.martini3.ff file from [here](https://github.com/marrink-lab/polyply_1.0/blob/master/polyply/data/martini3/PEO.martini3.ff), write the link file below to link.ff and then run:
```
polyply gen_params -f PPO.ff PEO.martini3.ff link.ff -seq OHter:1 PEO: 10 PPO:10 -o PEO_b_PPO.itp -name PPO
```
**The link parameters are only guesses based on the homopolymers.** @ricalessandri can get the proper parameters for you or at least show you how to do that. Coordinate generation should just work as usual. 

PPO file
```
[ moleculetype ]
; name nexcl.
PO        1
;
[ atoms ]
1   SN2r     1   PO   MOX   1   0.000  45

[ link ]
resname "PO"
[ bonds ]
; back bone bonds
MOX  >MOX     1   0.43   8000 {"comment": "MOX_MOX"}

[ link ]
resname "PO"
[ angles ]
MOX  >MOX >>MOX    2   125.00  100 {"comment": "MOX_MOX_MOX", "version":"1"}
```
link file
```
[ link ]
resname "PO|PEO"
[ bonds ]
EC >MOX 1   0.395  7500

[ link ]
resname "PO|PEO"
[ angles ]
EC >EC >>MOX 1 124 90

[ link ]
resname "PO|PEO"
[ angles ]
EC >MOX >>MOX  1  124 90
```

**Reply by shakhawath** [reply]
Hello @fgrunewald, Thanks for your detailed reply and the links for your thesis. I have started working with the homopolymers and it seems like they will work fine. It would be great if I could also get some more information about the link parameters between the PEO and PPO. However, with the values that you have here now, I can probably check the angle and bond distribution of the linked beads and compare them with previous literature data.  

Thanks.

**Reply by ricalessandri** [reply]
hi @shakhawath, sorry for the slow follow up. 

To get the proper links between PEO and PPO, you would follow the same workflow as explained in [this tutorial](https://github.com/ricalessandri/Martini3-small-molecules/blob/main/tutorials/M3tutorials--parameterizing-a-new-small-molecule.md). I'd recommend parametrizing a "small molecule" that represents a polymer fragment of 3 repeat units of PEO and 3 repeat units of PPO.

PEO_1-PEO_2-PEO_3-PPO_1-PPO_2-PPO_3

as this would allow you to get all the bonds (PEO_3-PPO_1), angles (PEO_2-PEO_3-PPO_1, PEO_3-PPO_1-PPO_2), and dihedrals (PEO_1-PEO_2-PEO_3-PPO_1, PEO_2-PEO_3-PPO_1-PPO_2, PEO_3-PPO_1-PPO_2-PPO_3) terms you may possibly need for the PEO-PPO link. Let me know if you have any questions on this!

**Reply by shakhawath** [reply]
Thank you very much. I can do that and check the parameter values.
